pilot/pkg/xds/delta.go

```diff
@@ -389,12 +394,17 @@ func (s *DiscoveryServer) pushDeltaXds(con *Connection,
 	t0 := time.Now()
 
 	originalW := w
-	// If subscribe is set, client is requesting specific resources. We should just generate the
+	// If delta is set, client is requesting new resources or removing old ones. We should just generate the
 	// new resources it needs, rather than the entire set of known resources.
-	if subscribe != nil {
+	// Note: we do not need to account for unsubscribed resources as these are handled by parent removal;
```

Comment:
Is this true of delta variant? Based on the xds link (the delete resources section you shared) this is the behaviour for sotW only.
May be just comment issue?

Reply:
Strange. We should update envoy docs then